src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+    this.shape = 'triangle';
+
+    const sides: number[] = [a, b, c].sort(
+      (i: number, j: number): number => i - j,
+    );
+
+    if (sides[0] <= 0) {
+      throw new Error('Side size <= 0');
+    } else if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error('Wrong side size');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+  color: 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+    this.color = color;
 
-export class Rectangle implements Figure {}
+    if (radius <= 0) {
+      throw new Error('radius must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public width: number,
+    public height: number,
+  ) {
+    this.shape = 'rectangle';
+    this.color = color;
+
+    if (width <= 0 || height <= 0) {
+      throw new Error('Invalid width or heigh');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. Instead of 'Invalid width or heigh', provide a more descriptive message, such as 'Rectangle width and height must be greater than 0'.